rustfmt-core/rustfmt-lib/src/config.rs

```diff
@@ -213,101 +247,108 @@ impl Config {
-        fn resolve_project_file(dir: &Path) -> Result<Option<PathBuf>, Error> {
+        fn resolve_project_files(dir: &Path) -> Result<Option<Vec<PathBuf>>, Error> {
             let mut current = if dir.is_relative() {
                 env::current_dir()?.join(dir)
             } else {
                 dir.to_path_buf()
             };
 
             current = dunce::canonicalize(current)?;
+            let mut paths = Vec::new();
 
             loop {
-                match get_toml_path(&current) {
-                    Ok(Some(path)) => return Ok(Some(path)),
-                    Err(e) => return Err(e),
-                    _ => (),
-                }
+                let current_toml_path = get_toml_path(&current)?;
+                paths.push(current_toml_path);
 
                 // If the current directory has no parent, we're done searching.
                 if !current.pop() {
                     break;
                 }
             }
 
+            // List of closest -> most distant rustfmt config from the current directory.
+            let config_paths: Option<Vec<_>> = paths.into_iter().filter(|p| p.is_some()).collect();
+            let has_paths = config_paths.as_ref().and_then(|paths| {
+                if paths.is_empty() {
+                    None
+                } else {
+                    Some(paths.len())
+                }
+            });
+            if has_paths.is_some() {
+                return Ok(config_paths);
+            }
+
             // If nothing was found, check in the home directory.
             if let Some(home_dir) = dirs::home_dir() {
                 if let Some(path) = get_toml_path(&home_dir)? {
-                    return Ok(Some(path));
+                    return Ok(Some(vec![path]));
                 }
             }
 
             // If none was found ther either, check in the user's configuration directory.
             if let Some(mut config_dir) = dirs::config_dir() {
                 config_dir.push("rustfmt");
                 if let Some(path) = get_toml_path(&config_dir)? {
-                    return Ok(Some(path));
+                    return Ok(Some(vec![path]));
                 }
             }
 
             Ok(None)
         }
 
-        match resolve_project_file(dir)? {
+        match resolve_project_files(dir)? {
             None => Ok((Config::default(), None)),
-            Some(path) => Config::from_toml_path(&path).map(|config| (config, Some(path))),
-        }
-    }
+            Some(paths) => {
+                // If this branch is hit, there must be at least one config file available.
+                let most_local_path = &paths[0];
+                if let Ok(partial_config) = PartialConfig::from_toml_path(most_local_path) {
+                    let config =
+                        Config::default().fill_from_parsed_config(partial_config, most_local_path);
+                    if config.version() == Version::One {
```

Comment:
We don't need this version gate anymore on the master branch, sorry if you have seen outdated documentation about this.

Reply:
Sounds good. Why is the gate not needed? Will releases be cut differently?

Reply:
Yes, we are planning to manage breaking changes by separating branches.

Reply:
fixed